airbyte-integrations/connectors/source-jira/source_jira/components/paginators.py

```diff
@@ -0,0 +1,68 @@
+#
+# Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+#
+
+import urllib.parse as urlparse
+from dataclasses import InitVar, dataclass
+from typing import Any, Dict, List, Mapping, Optional, Union
+from urllib.parse import parse_qsl
+
+import requests
+from airbyte_cdk.sources.declarative.decoders.decoder import Decoder
+from airbyte_cdk.sources.declarative.decoders.json_decoder import JsonDecoder
+from airbyte_cdk.sources.declarative.interpolation.interpolated_string import InterpolatedString
+from airbyte_cdk.sources.declarative.requesters.paginators.strategies.pagination_strategy import PaginationStrategy
+from airbyte_cdk.sources.declarative.types import Config, Record
+
+
+@dataclass
+class UrlPaginationStrategy(PaginationStrategy):
+    """
+    Custom component for cases when there is no information in the response body to match the current cursor position.
+    """
+
+    config: Config
+    parameters: InitVar[Mapping[str, Any]]
+    cursor_field_name: Union[InterpolatedString, str]
+    page_size: Optional[int] = None
+    decoder: Decoder = JsonDecoder(parameters={})
+
+    def __post_init__(self, parameters: Mapping[str, Any]) -> None:
+        if isinstance(self.cursor_field_name, str):
+            self._cursor_field_name = InterpolatedString.create(self.cursor_field_name, parameters=parameters)
+        else:
+            self._cursor_field_name = self.cursor_field_name
+
+    @property
+    def initial_token(self) -> Optional[Any]:
+        return None
+
+    def next_page_token(self, response: requests.Response, last_records: List[Record]) -> Optional[Any]:
```

Comment:
I'm confused by how this works. Why can't we use a standard limit offset paginator and increment the offset by the size of the the pages?

Reply:
@roman-yermilov-gl do you have an example of an API pagination we can't support? I'd expect CursorPaginator to cover string-based cursor but it's entirely possible we have gaps in our coverage

Reply:
I just was talking about a case when  the next page token is a part of url string:
```
{
    "next": {"nextUrl": "https://example.com?cursor=<the-cursor>"},
    "data": [{ object }]
}
```
In this example I think we can't specify `next.nextUrl` as a cursor field path